src/overlays/actors/ovl_En_Zod/z_en_zod.c

```diff
@@ -28,62 +57,592 @@ const ActorInit En_Zod_InitVars = {
+        }
+
+        switch (i) {
+            case 0:
+                if (((this->curAnimIndex == ENZOD_ANIM_PLAYING_LENTO) ||
+                     (this->curAnimIndex == ENZOD_ANIM_PLAYING_ANDANTINO)) &&
+                    ((s32)this->skelAnime.curFrame == 1)) {
+                    this->drumScaleVels[i] = 0.1f;
+                }
+                break;
+
+            case 2:
+                if (((this->curAnimIndex == ENZOD_ANIM_PLAYING_LENTO) && ((s32)this->skelAnime.curFrame == 19)) ||
+                    ((this->curAnimIndex == ENZOD_ANIM_PLAYING_ANDANTINO) && ((s32)this->skelAnime.curFrame == 8))) {
+                    this->drumScaleVels[i] = 0.1f;
+                }
+                break;
+
+            case 3:
+                switch (this->curAnimIndex) {
+                    case 3:
+                        switch ((s32)this->skelAnime.curFrame) {
+                            case 1:
+                            case 7:
+                            case 12:
+                            case 19:
+                                this->drumScaleVels[i] = 0.03f;
+                                break;
+                        }
+                        break;
+
+                    case 4:
+                        if ((s32)this->skelAnime.curFrame == 1) {
+                            this->drumScaleVels[i] = 0.1f;
+                        }
+                        break;
+                }
+                break;
+
+            case 4:
+                if ((this->curAnimIndex == ENZOD_ANIM_PLAYING_ANDANTINO) && ((s32)this->skelAnime.curFrame == 19)) {
+                    this->drumScaleVels[i] = 0.15f;
+                }
+
+                break;
+
+            default:
+                break;
+        }
+    }
+}
+
+void func_80BAF7CC(EnZod* this, PlayState* play) {
+    EnZod_UpdateAnimations(this);
+    switch (Message_GetState(&play->msgCtx)) {
+        case TEXT_STATE_CHOICE:
+            if (Message_ShouldAdvance(play) && (play->msgCtx.currentTextId == 0x121F)) {
+                switch (play->msgCtx.choiceIndex) {
+                    case 0:
+                        func_8019F208();
+                        func_80151938(play, 0x1220); // There's a rumor that the ocean is getting...
+                        break;
+
+                    case 1:
+                        func_8019F230();
+                        func_80151938(play, 0x1223); // Oh, yeah, that's right. There aren't any secrets...
+                        break;
+                }
+            }
+            break;
+
+        case TEXT_STATE_5:
+            if (Message_ShouldAdvance(play)) {
+                switch (play->msgCtx.currentTextId) {
+                    case 0x121A:
+                    case 0x121B:
+                    case 0x121C:
+                    case 0x1220:
+                    case 0x1221:
+                    case 0x1227:
+                        func_80151938(play, (play->msgCtx.currentTextId + 1) & 0xFFFF);
```

Comment:
this function already takes a `u16`, is the `& 0xFFFF` part required?
```suggestion
                        func_80151938(play, (play->msgCtx.currentTextId + 1));
```
same comment for the other calls to this function